packages/odyssey-react/src/utils/useSharedRef.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { ForwardedRef, RefObject, useEffect, useRef } from "react";
+
+export function useSharedRef<T>(ref?: ForwardedRef<T>): RefObject<T> {
+  const internalRef = useRef<T>(null);
+
+  useEffect(() => {
+    if (!ref) {
+      return;
+    }
+    if (typeof ref === "function") {
+      ref(internalRef.current);
+    } else {
+      ref.current = internalRef.current;
+    }
+  }, [ref, internalRef]);
+
+  return internalRef;
+}
```

Comment:
I believe we already have some places in our code base where we can use this util! I think it was `Checkbox` perhaps a fast follow PR to refactor there

Reply:
Yes, I just wasn't sure what the refactor preference was for Odyssey? I'll make a follow-up PR to sprinkle in `useSharedRef` in applicable places?